Show spotlight in other slidev tabs

marob opened the pull request:
When using slidev presenter & presentation mode in 2 tabs, spotlight is only visible on current tab.

Here is a possible implementation.

marob committed:
Show spotlight in other slidev tabs

When using slidev presenter & presentation mode in 2 tabs, spotlight is only visible on current tab.

Here is a possible implementation.

tonai commented:
Currently I have an error when starting the dev server with `npm run serve`:
```
 ERROR  Failed to compile with 1 error                                                                                                                                                                     09:12:35

This dependency was not found:

* server-reactive:nav in ./node_modules/cache-loader/dist/cjs.js??ref--14-0!./node_modules/babel-loader/lib!./node_modules/ts-loader??ref--14-2!./node_modules/cache-loader/dist/cjs.js??ref--1-0!./node_modules/vue-loader-v16/dist??ref--1-1!./src/spotlight.vue?vue&type=script&setup=true&lang=ts
```

tonai commented:
Importing `server-reactive:nav` is a feature from `vite-plugin-vue-server-ref` which is a dependency of `@slidev/cli`.
The problem is that this plugin is a Vue and not a Slidev only component.
Maybe we should create another plugin `slidev-component-spotlight` that will import this component and do the `slidev` stuff here.

marob committed:
Fixes/improvements:

- activating spotlight was changing page (to 1st one)
- import from @slidev/client/env instead of defining another server-reactive
- add throttling
- only send spotlight position to server if in presentation mode (to "copy" the spotlight from presenter to play mode)

marob committed:
Hide presenter cursor when spotlight is active

tonai commented:
Slidev part moved to https://github.com/Smile-SA/slidev-component-spotlight
